modules/network/multivpc/variables.tf

```diff
@@ -0,0 +1,186 @@
+  default     = 8896
+}
+
+variable "network_routing_mode" {
+  type        = string
+  default     = "REGIONAL"
+  description = "The network dynamic routing mode"
+
+  validation {
+    condition     = contains(["GLOBAL", "REGIONAL"], var.network_routing_mode)
+    error_message = "The network routing mode must either be \"GLOBAL\" or \"REGIONAL\"."
+  }
+}
+
+variable "network_description" {
+  type        = string
+  description = "An optional description of this resource (changes will trigger resource destroy/create)"
+  default     = ""
+}
+
+variable "ips_per_nat" {
+  type        = number
+  description = "The number of IP addresses to allocate for each regional Cloud NAT (set to 0 to disable NAT)"
+  default     = 2
+}
+
+variable "delete_default_internet_gateway_routes" {
+  type        = bool
+  description = "If set, ensure that all routes within the network specified whose names begin with 'default-route' and with a next hop of 'default-internet-gateway' are deleted"
+  default     = false
+}
+
+variable "enable_iap_ssh_ingress" {
+  type        = bool
+  description = "Enable a firewall rule to allow SSH access using IAP tunnels"
+  default     = true
+}
+
+variable "enable_iap_rdp_ingress" {
+  type        = bool
+  description = "Enable a firewall rule to allow Windows Remote Desktop Protocol access using IAP tunnels"
+  default     = false
+}
+
+variable "enable_iap_winrm_ingress" {
+  type        = bool
+  description = "Enable a firewall rule to allow Windows Remote Management (WinRM) access using IAP tunnels"
+  default     = false
+}
+
+variable "enable_internal_traffic" {
+  type        = bool
+  description = "Enable a firewall rule to allow all internal TCP, UDP, and ICMP traffic within the network"
+  default     = true
+}
+
+variable "extra_iap_ports" {
+  type        = list(string)
+  description = "A list of TCP ports for which to create firewall rules that enable IAP for TCP forwarding (use dedicated enable_iap variables for standard ports)"
+  default     = []
+}
+
+variable "allowed_ssh_ip_ranges" {
+  type        = list(string)
+  description = "A list of CIDR IP ranges from which to allow ssh access"
+  default     = []
+
+  validation {
+    condition     = alltrue([for r in var.allowed_ssh_ip_ranges : can(cidrhost(r, 32))])
+    error_message = "Each element of var.allowed_ssh_ip_ranges must be a valid CIDR-formatted IPv4 range."
+  }
+}
+
+variable "firewall_rules" {
+  type        = any
+  description = "List of firewall rules"
+  default     = []
+}
+
+variable "network_template" {
+  type = object({
+    network            = optional(string)
+    subnetwork         = optional(string)
+    subnetwork_project = optional(string)
+    network_ip         = optional(string)
+    nic_type           = string
+    stack_type         = string
+    queue_count        = optional(string)
+    access_config = optional(list(object({
```

Comment:
Please add the field public_ptr_domain_name to both access_config fields. This make it compatible with both Slurm modules (where the field gets ignored but we should add it) and vm-instance (where the field is required).

```hcl
     access_config = optional(list(object({
-      nat_ip       = string
-      network_tier = string
+      nat_ip                 = string
+      network_tier           = string
+      public_ptr_domain_name = string
     })))
     ipv6_access_config = optional(list(object({
-      network_tier = string
+      network_tier           = string
+      public_ptr_domain_name = string
     })))
```